File: Practice/21/C++/21.cpp
// 21.cpp : Этот файл содержит функцию "main". Здесь начинается и заканчивается выполнение программы.
//

#include <iostream>
using namespace std;

double BMI(double weight, double height)    //функция подсчета BIM
{
    return ((weight / (pow(height, 2))) * pow(10, 4));
}
void printBMI(double BMI) //Функция выбора категории
{
    if (BMI < 18.5)
    {
        cout << "Underweight\n";
    }
    else if (18.5 <= BMI && BMI < 25)
    {
        cout << "Normal\n";
    }
    else if (25 <= BMI && BMI < 30)
    {
        cout << "Overweight\n";
    }
    else
    {
        cout << "Obesity\n";
    }
}
int main()
{ 
    setlocale(0, "");
    double ves, rost, n;
    int v = 0;
    do {
            cout << "Введите вес и рост:\n";
            cin >> ves >> rost;
            if (ves != 0 && rost != 0)  //проверка на положительность чисел
            {
                n = BMI(ves, rost); //вызов функций
                printBMI(n);
                cout << "закрыть программу? 1-да ";
                cin >> v;
            }
            else
            {
                cout<<"Введите положительные значения: \n";
            }
    } while (v != 1);
}



       

File: Practice/21/Python/_21.py
#!/usr/bin/env python3
def main():
    ves, rost = input("Введите ваш рост и вес: \n").split()
    ves = float(ves)
    rost = float(rost)
    if ves != 0 and rost != 0:
        def bmi (weight, height):       #Функция вычисляющая BMI
                return (weight / (height**2))*(10**4) 
        n = bmi(ves, rost)
        def print_bmi(bmi):           #Функция выводящая значение
            if bmi < 18.5:
                print("Underweight")
            elif 18.5 <= bmi < 25:
                print("Normal")
            elif 25 <= bmi < 30:
                print("Overweight")
            else:
                print("Obesity")
        print_bmi(n) # Вызов функции
        vixod = input("закрыть программу? y-да: \n")
        if vixod == "y":
            quit(0)
    else: print("Введите положительные значения: ")
if __name__ == '__main__':
    while True:  
        main()
